service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseRoleService.java

```diff
@@ -73,6 +81,56 @@ private void grantCaseAccess(String caseId, Set<String> users, Set<CaseRole> rol
         }
     }
 
+    public void grantCaseAssignmentToUser(String caseId, String user, Set<CaseRole> roles) {
+        grantCaseAssignmentAccess(caseId, Set.of(user), roles);
+        log.info("User {} granted {} to case {}", user, roles, caseId);
+    }
+
+    public void grantCaseAssignmentToLocalAuthority(String caseId, String localAuthority, Set<CaseRole> roles,
+                                            Set<String> excludeUsers) {
+        Set<String> localAuthorityUsers = getUsers(caseId, localAuthority, excludeUsers, roles);
+        grantCaseAccess(caseId, localAuthorityUsers, roles);
+        log.info("Users {} granted {} to case {}", localAuthorityUsers, roles, caseId);
+    }
+
+    private void grantCaseAssignmentAccess(String caseId, Set<String> users, Set<CaseRole> roles) {
+        final Set<String> usersGrantedAccess = Sets.newConcurrentHashSet();
+
+        try {
+            final String userToken = idam.getAccessToken(userConfig.getUserName(), userConfig.getPassword());
+            final String serviceToken = authTokenGenerator.generate();
+            List<CaseAssignedUserRoleWithOrganisation> caseAssignedRoles = new ArrayList<>();
+            AddCaseAssignedUserRolesRequest addCaseRequest = new AddCaseAssignedUserRolesRequest();
+
+            users.stream().sequential()
+                .forEach(userId -> {
+                    try {
+                        CaseAssignedUserRoleWithOrganisation caseUserRole = new CaseAssignedUserRoleWithOrganisation();
+                        // It has been decided that this will be tested in preview with blank org id first.
+                        caseUserRole.setOrganisationId("");
```

Comment:
i think we should set organisation id if user is in MO, you can use OrganisationService.findOrganisation

Reply:
It has been decided that when CCD enables us on demo, they want to try it out with blank organisation name